desc/optimize/_constraint_wrappers.py

```diff
@@ -502,6 +549,26 @@ def build(self, use_jit=None, verbose=1):  # noqa: C901
             self._scalar = False
 
         self._set_eq_state_vector()
+
+        # map from eq c to full c
+        self._dimc_per_thing = [t.dim_x for t in self.things]
```

Comment:
This is just saying that every optimizable param of things other than eqs is fair game to be changed, but for the equilibrium the inputs c are actually different than dim_x ?

Reply:
yeah the equilibrium only considers the boundary dofs, other objects have everything counted